### ISSUE TITLE
Setting "Show Description" toggle to False in CPE doesn't save attribute to Flow Properties

### ISSUE THREAD

**Issue by jkranz-rk** [jkranz-rk/RecordTypePicker#18]
When configuring the Record Type Picker with the CPE, toggling the Show Description option to "Off/False" will appear to save while you are in the Screen configuration, but the value on the component's property doesn't get saved to the new value, so when you return the Screen, or run the Flow, the descriptions will continue to show.

Expected result:

Toggling the Show Description setting to off should persist

**Comment by ericrsmith35** [comment]
@jkranz-rk I built this reusable component to help with saving checkbox values in CPEs.  https://unofficialsf.com/checkbox-component-for-custom-property-editors/

**Comment by jkranz-rk** [comment]
@ericrsmith35  interesting and very neat! In this case, the bug was a simple mistake on my part regarding how default value for the core LWC property was getting set and publishing a patch this morning, but I'll keep this in my back pocket, thank you!

**Comment by jkranz-rk** [comment]
In the future I'm hoping #15 (automated jest/UI tests) will keep this from happening 😅

**Comment by jkranz-rk** [comment]
@ericrsmith35 As it turns out- issue was a bit different than I thought. So then I took a closer look at the link / solution you shared, and I thought "oh well maybe that _is_ it!" But I was hesitant to add an extra string `@api` property to the core component, so I dug a bit deeper.

What I found was that the `inputVariables` interface provides Boolean values to the CPE as a string value: `'$GlobalConstant.True'` and `'$GlobalConstant.False'`. These values _look like_ but are _not_ merge syntax reference values. The `configuration_editor_input_value_changed` event for Boolean properties _also_ expects these strings instead of `True` or `False` or even Truthy/Falsey values.

To address this, instead of an extra string property that lives alongside the boolean property, what I did was transform the string to boolean any time the CPE reads a Boolean value from incoming `inputVariables`, and then transform it back to a string when reporting changes back to the builder with the change event.

It seems silly to have to do this, I feel as though the builder should be doing this transformation. But! This approach seems to be consistently and reliably working so far. I'll be doing a bit more testing and publishing an updated package soon with this fix (hopefully today).
